---
layout: archive
title: Publications
author_profile: true
permalink: /publications
sidebar:
  - title: ""
  - text: |
      ---
      A potentially more complete and updated version of my bibliography can be found at the following links:

      [ORCID: 0000-0001-6406-8563](https://orcid.org/0000-0001-6406-8563)
      
      [Google Scholar](https://scholar.google.com/citations?user=wlX37YwAAAAJ&hl=en)

      [Pubmed](https://www.ncbi.nlm.nih.gov/pubmed/?term=vierstra+j%5Bau%5D)
---

Reverse chronological order, * denotes equal contribution

---

Georgolopoulos G, Iwata M, Psatha N, Yiangou M, ___Vierstra J___ (2019). Unbiased phenotypic identification of functionally distinct hematopoietic progenitors. *J Biol Res (Thessalon)*, (26)4.

Breeze CE, Reynolds AP, van Dongen J, Dunham I, Lazar J, Neph S, ___Vierstra J___, Bourque G, Teschendorff AE, Stamatoyannopoulos JA, Beck S (2019). eFORGE v2.0: updated analysis of cell type-specific signal in epigenomic data. *Bioinformatics*, in press.

Sieber KB, Batorsky A, Siebenthall KT, Hudkins KL, ___Vierstra J___, Sullivan S, Sur A, McNulty M, Sandstrom R, Reynolds A, Bates D, Diegel M, Dunn D, Nelson J, Buckley M, Kaul R, Sampson MG, Himmelfarb J, Alpers CE, Waterworth D, Akilesh S (2019). Integrated Functional Genomic Analysis Enables Annotation of Kidney Genome-Wide Association Study Loci. *J Am Soc Nephrol*, 30(3):421-41.

Siebenthall KT, Miller CP, ___Vierstra J___, Mathieu J, Tretiakova M, Reynolds A, Sandstrom R, Rynes E, Haugen E, Johnson A, Nelson J, Bates D, Diegel M, Dunn D, Frerker M, Buckley M, Kaul R, Zheng Y, Himmelfarb J, Ruohola-Baker H, Akilesh S (2019). Integrated epigenomic profiling reveals endogenous retrovirus reactivation in renal cell carcinoma. *EBioMedicine*, 41:427-442.

___Vierstra J___ & Stamatoyannopoulos (2016). Genomic footprinting. *Nat Methods*. 13(3):213–221. *Review*.

He X, Tillo D, ___Vierstra J___, Syed KS, Deng C, Ray JG, Stamatoyannopoulos JA, Fitzgerald PC, Vinson C (2015). Methylated cytosines mutate to transcription factor binding sites that drive tetrapod evolution. *Genome Biol Evol*, 7(11):3155-69.

Maurano MT, Haugen E, Sandstrom R, ___Vierstra J___, Shafer T and Stamatoyannopoulos JA (2015). Large-scale identification of functional variants impacting human transcription factor occupancy in vivo. *Nat Genetics*, 47(12):1393-401.

___\*Vierstra J___, \*Reik A, Chang KH, Stehling-Sun S, Zhou YY, Hinkley SJ, Paschon DE, Zhang L, Psatha N, Bendana YR, O’Neill CM, Song AH, Mich A, Liu P-Q, Lee G, Bauer DE, Holmes MC, Orkin SH, Papayannopoulou T, Stamatoyannopoulos G, Rebar EJ, Gregory PD, Urnov FD, Stamatoyannopoulos JA (2015). Functional footprinting of regulatory DNA. *Nat Methods*, 12(10):927-930.

Mayer A, di Lulio J, Maleri S, Eser U, ___Vierstra J___, Reynolds A, Sandstrom R, Stamatoyannopoulos JA, Churchman LS (2015). Native elongating transcript sequencing reveals human transcriptional activity at nucleotide resolution. Cell, 161(3):541-554.

Wilken MS, Brzezinski JA, La Torre A, Siebenthall K, Thurman R, Sabo P, Sandstrom RS, ___Vierstra J___, Canfield TK, Hansen RS, Bender MA, Stamatoyannopoulos J, Reh TA (2015). DNase I hypersensitivity analysis of the mouse brain and retina identifies region-specific regulatory elements. Epigenetics Chromatin, 8(1):8.

Stergachis AB, Neph S, Sandstrom R, Haugen E, Reynolds AP, Zhang M, Byron R, Canfield T, Stelhing-Sun S, Lee K, Thurman RE, Vong S, Bates D, Neri F, Diegel M, Giste E, Dunn D, ___Vierstra J___, Hansen RS, Johnson AK, Sabo PJ, Wilken MS, Reh TA, Treuting PM, Kaul R, Groudine M, Bender MA, Borenstein E, Stamatoyannopoulos JA. (2014). Conservation of trans-acting circuitry during mammalian regulatory evolution. Nature, 515(7527):355-364.

Mouse ENCODE Project Consortium, \*Yue F, \*Cheng Y, \*Breschi A, ___\*Vierstra J___, \*Wu W, \*Ryba T, \*Ma Z, \*Davis C, \*Pope BD, \*Shen Y et al. (2014). An integrated and comparative encyclopedia of DNA elements in the mouse genome. Nature, 515(7527):365-370.


___Vierstra J___, Rynes E, Sandstrom R, Thurman RE et al. (2014). Mouse regulatory DNA landscapes reveal global principles of cis-regulatory evolution. Science, 346(6212):1007-1012.

___Vierstra J___, Wang H, John S, Sandstrom R, Stamatoyannopoulos JA. (2014). Coupling transcription factor occupancy to nucleosome architecture with DNase-FLASH. Nat Methods, 11(1):66–72.
Commentary: Raj and McVicker (2014). The genome shows its sensitive side. Nat Methods, 11(1):39-40.

Kamran SC, Lessard S, Xu J, Fujiwara Y, Lin C, Shao Z, Canver MC, Smith EC, Pinello L, Sabo PJ, ___Vierstra J___, Voit RA, Yuan GC, Porteus MH, Stamatoyannopoulos JA, Lettre G, Orkin SH. (2013). Fine-Mapping and Genome Editing Reveal An Essential Erythroid Enhancer At The HbF-Associated BCL11A Locus. Blood, 122 (21):437-437.

Bauer DE, Kamran SC, Lessard S, Xu J, Fujiwara Y, Lin C, Shao Z, Canver MC, Smith EC, Pinello L, Sabo PJ, ___Vierstra J___, Voit RA, Yuan GC, Porteus MH, Stamatoyannopoulos JA, Lettre G, Orkin SH. (2013). An Erythroid Enhancer of BCL11A Subject to Genetic Variation Determines Fetal Hemoglobin Level. Science, 342(6155):253-257.

John S, Sabo PJ, Canfield TK, Lee K, Vong S, Weaver M, Wang H, ___Vierstra J___, Reynolds AP, Thurman RE, Stamatoyannopoulos JA. (2013). Genome-scale mapping of DNase I hypersensitivity. Curr Protoc Mol Biol., Chapter 27: Unit 21.27.

Vernot B, Stergachis AB, Maurano MT, ___Vierstra J___, Neph S, Thurman RE, Stamatoyannopoulos JA, Akey JM. (2012). Personal and population genomics of human regulatory variation. Genome Res., 22(9):1689-97.

\*Neph S, ___\*Vierstra J___, \*Stergachis AB, \*Reynolds AP, Haugen E, Vernot B, Thurman RE, John S, Sandstrom R, Johnson AK, Maurano MT, Humbert R, Rynes E, Wang H, Vong S, Lee K, Bates D, Diegel M, Roach V, Dunn D, Neri J, Schafer A, Hansen RS, Kutyavin T, Giste E, Weaver M, Canfield T, Sabo P, Zhang M, Balasundaram G, Byron R, MacCoss MJ, Akey JM, Bender MA, Groudine M, Kaul R, Stamatoyannopoulos JA. (2012). An expansive human regulatory lexicon encoded in transcription factor footprints. Nature, 489(7414):83-90.

Thurman RE, Rynes E, Humbert R, ___Vierstra J___, Maurano MT, Haugen E, Sheffield NC, Stergachis AB, Wang H, Vernot B, Garg K, John S, Sandstrom R, Bates D, Boatman L, Canfield TK, Diegel M, Dunn D, Ebersol AK, Frum T, Giste E, Johnson AK, Johnson EM, Kutyavin T, Lajoie B, Lee BK, Lee K, London D, Lotakis D, Neph S, Neri F, Nguyen ED, Qu H, Reynolds AP, Roach V, Safi A, Sanchez ME, Sanyal A, Shafer A, Simon JM, Song L, Vong S, Weaver M, Yan Y, Zhang Z, Zhang Z, Lenhard B, Tewari M, Dorschner MO, Hansen RS, Navas PA, Stamatoyannopoulos G, Iyer VR, Lieb JD, Sunyaev SR, Akey JM, Sabo PJ, Kaul R, Furey TS, Dekker J, Crawford GE, Stamatoyannopoulos JA.(2012). The accessible chromatin landscape of the human genome. Nature, 489(7414):75-82.

ENCODE Project Consortium, Dunham I, Kundaje A, et al. [410 authors]. (2012). An integrated encyclopedia of DNA elements in the human genome. Nature, 489(7414):57-74.

Neph S, Kuehn MS, Reynolds AP, Haugen E, Thurman RE, Johnson AK, Rynes E, Maurano MT, ___Vierstra J___, Thomas S, Sandstrom R, Humbert R, Stamatoyannopoulos JA. (2012). BEDOPS: high-performance genomic feature operations. Bioinformatics, 28(14):1919-20.

