---
layout: archive
author_profile: true
---

I am an investigator at the Altius Institute for Biomedical Sciences. My research focuses on deciphering the structure-function relationship of chromatin. Some of my main areas of interest are:

- **Mapping chromatin structure & function using nucleases**

  Specific and non-specific endonucleases have arguably been the most important tools in modern genomic research (*e.g.*, restriction nucleases or more recently genome-editing reagents). We develop methods that leverage nucleases to create *in vivo* nucleotide-resolution [maps of transcription factor occupancy](https://www.nature.com/articles/s41586-020-2528-x) and [function](https://www.nature.com/articles/nmeth.3554) within the genome of primary cells, tissue types and disease states.

- **Understanding gene regulation through the lens of human genetics**

  The regulation of gene expression by *cis*-regulatory DNA elements can occur over tremendous distances. Linking individual distal regulatory elements to their cognate gene(s) remains a significant challenge. We leverage naturally occuring human regulatory genetic variation and make use of allelic imbalance in epigenomics and gene expression data across large cohorts of distinct individuals and cell-types to gain mechanistic insights into how genes are normally regulated and disregulated in disease.

- **Cis-regulatory encoding of cell identity and fate potential**

  Lineage commitment and differentiation is driven by the concerted action of master transcriptional regulators at their target chromatin sites. We use erythropoiesis, megakaryopoiesis and myelopoiesis to investigate how the temporal interplay of transcription factors with chromatin determines cell fate potential.

Prior to Altius, I did both doctoral and postdoctoral training at the University of Washington. I obtained my *BSc* in Genetics and Computer Science at the University of Wisconsin.